OpenRA.Game/Network/UnitOrders.cs

```diff
@@ -81,25 +81,30 @@ internal static void ProcessOrder(OrderManager orderManager, World world, int cl
 					{
 						var client = orderManager.LobbyInfo.ClientWithIndex(clientId);
 
-						if (client != null)
+						if (client == null)
+							break;
+
+						if (world == null)
 						{
-							if (world == null)
-							{
-								if (orderManager.LocalClient != null && client.Team == orderManager.LocalClient.Team)
-									Game.AddChatLine(client.Color, "[Team] " + client.Name, order.TargetString);
-							}
-							else
-							{
-								var player = world.FindPlayerByClient(client);
-								if (player != null && player.WinState == WinState.Lost)
-									Game.AddChatLine(client.Color, client.Name + " (Dead)", order.TargetString);
-								else if ((player != null && world.LocalPlayer != null && player.Stances[world.LocalPlayer] == Stance.Ally) || (world.IsReplay && player != null))
-									Game.AddChatLine(client.Color, "[Team" + (world.IsReplay ? " " + client.Team : "") + "] " + client.Name, order.TargetString);
-								else if ((orderManager.LocalClient != null && orderManager.LocalClient.IsObserver && client.IsObserver) || (world.IsReplay  && client.IsObserver))
-									Game.AddChatLine(client.Color, "[Spectators] " + client.Name, order.TargetString);
-							}
+							if (orderManager.LocalClient != null && client.Team == orderManager.LocalClient.Team)
+								Game.AddChatLine(client.Color, "[Team] " + client.Name, order.TargetString);
+
+							break;
 						}
 
+						var player = world.FindPlayerByClient(client);
+						var isSpectator = orderManager.LocalClient != null && orderManager.LocalClient.IsObserver;
```

Comment:
`isLiving` and `isDead` refer to the client who sent the message, but `isSpectator` refers to the client reading the message. Is this a bug, or just unfortunate naming?